### PR TITLE
facility and entitlement columns and filters on staff page

### PR DESCRIPTION
**Pull request by wittejm**
also extract new `MultiSelect` component and `prettyLabel` util

https://user-images.githubusercontent.com/2104990/190725188-caa87ac8-a6b0-4789-b4fe-f4384b6ef26d.mov

### CODE REVIEW

**Comment by wittejm** [comment]
oh darn editing staff facilities/entitlements no longer is immediately reflected in the table. fixing....

**Comment by wittejm** [comment]
Ok fixed. In this case I'm confused why triggering the useEffect on data works while triggering useMemo on data which updates staff which should trigger the useEffect.... doesn't work.  🤷

File: src/components/System/SystemStaff.tsx [hunk 1]
```diff
@@ -28,15 +52,37 @@ const SystemStaff = () => {
...
   const [isAddStaffModalOpen, setIsAddStaffModalOpen] = useState(false);
   const [editingStaff, setEditingStaff] = useState<Staff | null>(null);
+  const [filters, setFilters] = useState<Filters>({
+    facilities: [],
+    entitlements: [],
+  });
+
+  const [filteredStaff, setFilteredStaff] = useState<Staff[]>();
+  useEffect(() => {
+    setFilteredStaff(
```
Comment (cainlevy) [inline src/components/System/SystemStaff.tsx:68]: looks like a `const filteredStaff = useMemo(...)`

File: src/utils/prettyLabel.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  [FacilityService.Call]: "call",
+  [FacilityService.Emessaging]: "messaging",
+  [FacilityService.InPersonContact]: "in person (with contact)",
+  [FacilityService.InPersonNoContact]: "in person (no contact)",
+  [FacilityService.VoiceCall]: "voice call",
+};
+
+export const prettyLabel = (
+  thing: Entitlement | FacilityService | string,
```
Comment (cainlevy) [inline src/utils/prettyLabel.ts:21]: presumably this works as long as our enums don't collide?

**Review by cainlevy — APPROVED** [review]

File: src/components/System/SystemStaff.tsx (hunk 1 above)
Comment (cainlevy) [reply src/components/System/SystemStaff.tsx:68]: oh you commented about this while i was reviewing. hmm. i don't have any immediate ideas.

File: src/utils/prettyLabel.ts (hunk 2 above)
Comment (wittejm) [reply src/utils/prettyLabel.ts:21]: Yep. If that becomes an issue, we can rethink it; I considered it a low-effort refactor

**Pull request merged**